tests/test_asyncio/test_commands.py

```diff
@@ -2999,6 +3001,32 @@ async def test_module_list(self, r: redis.Redis):
         for x in await r.module_list():
             assert isinstance(x, dict)
 
+    @pytest.mark.onlynoncluster
+    async def test_interrupted_command(self, r: redis.Redis):
+        """
+        Regression test for issue #1128:  An Un-handled BaseException
+        will leave the socket with un-read response to a previous
+        command.
+        """
+        ready = asyncio.Event()
+
+        async def helper():
+            with pytest.raises(asyncio.CancelledError):
+                # blocking pop
+                ready.set()
+                await r.brpop(["nonexist"])
+            # if all is well, we can continue.  The following should not hang.
+            await r.set("status", "down")
```

Comment:
Still tbd to validate that not only you don't "hang", but also that the next read response matches the command.